### PR TITLE
Fixes #29986 - Capsules shouldn't update hosts' Registered through f…

### PR DESCRIPTION
**Pull request by piyushtiw**
…acts on the Satellite server in a load-balanced configuration

### CODE REVIEW

**Comment by piyushtiw** [comment]
[test katello]

**Comment by chris1984** [comment]
Starting to test

**Comment by chris1984** [comment]
[test katello]

File: app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb [hunk 1]
```diff
@@ -268,8 +268,8 @@ def serials
     end
 
     def get_parent_host(headers)
-      hostnames = headers["HTTP_X_FORWARDED_SERVER"]
-      host = hostnames.split(",")[0] if hostnames
+      hostnames = headers["HTTP_X_FORWARDED_HOST"]
+      host = hostnames.split(/[\,,:]/)[0] if hostnames
```
Comment (jturel) [inline app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb:272]: What's the difference between your regex and the `split` approach?

Comment (jturel) [inline app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb:272]: May wanna add a `.strip` here to ensure any whitespace is omitted as well

Comment (jturel) [reply app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb:272]: Ah, ok! Can you include such an example in your unit tests? That'll make sure we don't break this later

**Comment by piyushtiw** [comment]
[test katello]

**Comment by jturel** [comment]
[test katello]

**Comment by chris1984** [comment]
Thanks @piyushtiw

**Pull request merged**